packages/frontend/core/src/modules/quicksearch/impls/docs.ts

```diff
@@ -57,23 +60,33 @@ export class DocsQuickSearchSession
       } else {
         const resolvedDoc = resolveLinkToDoc(query);
         const resolvedDocId = resolvedDoc?.docId;
-        const resolvedBlockId = resolvedDoc?.blockIds?.[0];
+        const params =
+          resolvedDoc && pick(resolvedDoc, ['mode', 'blockIds', 'elementIds']);
+        const resolvedBlockId = params?.blockIds?.[0];
 
         out = this.docsSearchService.search$(query).pipe(
           map(docs => {
-            if (
-              resolvedDocId &&
-              !docs.some(doc => doc.docId === resolvedDocId)
-            ) {
-              return [
-                {
-                  docId: resolvedDocId,
-                  score: 100,
-                  blockId: resolvedBlockId,
-                  blockContent: '',
-                },
-                ...docs,
-              ];
+            if (!resolvedDocId) {
+              return docs;
+            }
+
+            const doc = docs.find(doc => doc.docId === resolvedDocId);
+
+            if (resolvedBlockId) {
+              if (doc && doc.blockId !== resolvedBlockId) {
+                doc.blockId = resolvedBlockId;
+                doc.blockContent = '';
```

Comment:
please don't mutate input in `.map`

Reply:
The current search result may not match the `reslovedBlockId`.

Reply:
I know, but it is hard to know here whether or not the doc is cached in the search result.
Mutating it is dangerous.

Try something like this:

```ts
const idx = docs.findIndex(doc => doc.docId === resolvedDocId);
const doc = { ...doc };
doc.blockId = xxx;
doc.blockContent = 'yyy';
docs = docs.toSpliced(idx, 1, doc)
```

Reply:
TIL `toSpliced`